tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,571 @@
+---
+# vim: set ft=ansible:
+#
+#   This is a general test suite for rpm-ostree that exercises several of the
+# rpm-ostree subcommands.  These tests assume the atomic host is provisioned
+# with the latest version.  These tests are meant to be run in succession.
+#
+- name: rpm-ostree - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  roles:
+    # This playbook requires Ansible 2.2 and an Atomic Host
+    - role: ansible_version_check
+      avc_major: "2"
+      avc_minor: "2"
+      tags:
+        - ansible_version_check
+
+    # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+#####################################################################################
+#
+# rpm-ostree deploy version
+#   - validates deploying by version string
+#   - validates cleanup of rollback deployment
+#
+#####################################################################################
+
+- name: rpm-ostree - deploy by version
+  hosts: all
+  become: yes
+
+  tags:
+    - deploy_version
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - name: Get current refspec
+      shell: cat $(ostree admin --print-current-dir).origin | grep refspec | cut -d= -f2
+      register: refspec
+
+    - name: Get json ouput from rpm-ostree status
+      command: rpm-ostree status --json
+      register: ros_status
+
+    - name: Convert to json
+      set_fact:
+         ross_json: "{{ ros_status.stdout | from_json }}"
+
+    - name: Set current version
+      set_fact:
+         head_version: "{{ ross_json['deployments'][0]['version'] if ross_json['deployments'][0]['booted'] else ross_json['deployments'][1]['version'] }}"
+
+    - name: Pull last two commits
+      command: ostree pull --subpath /usr/share/rpm --depth=1 {{ refspec.stdout }}
```

Comment:
We switched to this for  CAHC and FAHC.  I was away when this change happened so I don't know the specifics.  It has something to do with commit history on the continuous streams.  @miabbott do you know offhand why we moved to this instead of commit-metadata-only?